zubhub_frontend/zubhub/src/api/api.js

```diff
@@ -854,7 +854,7 @@ class API {
    * @todo - describe method's signature
    */
   getProjects = ({ token, page }) => {
-    const url = page ? `projects/?${page}` : `projects/`;
+    const url = page ? `projects/?page=${page}&page_size=${PROJECTS_PAGE_SIZE}` : `projects/?page_size=${PROJECTS_PAGE_SIZE}`;
```

Comment:
OK, so I don't understand the goal of using `PROJECT_PAGE_SIZE` here in URI. This is mostly relevant for getting total number of pages you have to show in the pagination view isn't it? In that case, we can just decide on a fixed number, like 18 and put it in utils.js. Then in UI, we can adapt on how many paginations are we showing. eg

<---- [1, 2. 3, 4, 5, 6... ] of [256]   --->

As you reach number 6, the view should then auto adjust UI as:

<--- [1...6, 7, 8, 9, 10] of [256] ---> 

and so on.

Reply:
> This is mostly relevant for getting total number of pages you have to show in the pagination view isn't it?

Yes!

> In that case, we can just decide on a fixed number, like 18 and put it in utils.js.

The backend has a hard-coded value of `18` right now which will be treated as the default if we don't pass the `page_size` param. We could hard-code it in the frontend too like you suggest. But the reason I added this is to allow flexibility on the frontend side. Say we ever want to change the number of projects per page in the future; all we have to do is change this configuration on the frontend and it will just work. No need to make any changes on the backend at all. If we go with the hard-coded approach, we'd have to make changes on both ends. 

> In that case, we can just decide on a fixed number, like 18 and put it in utils.js. Then in UI, we can adapt on how many paginations are we showing. eg
> 
> <---- [1, 2. 3, 4, 5, 6... ] of [256] --->
> 
> As you reach number 6, the view should then auto adjust UI as:
> 
> <--- [1...6, 7, 8, 9, 10] of [256] --->
> 
> and so on.
> 
And yes, this one is already taken care of.

Reply:
If we accept the page size parameter we also have to validate the minimum and maximum values in the backend

Reply:
@aqsaaqeel I feel we should fix the page size parameter to 18 for now and not allow users to set it. We can introduce user configurable values later. For now, we want to keep it simple and navigable